src/App.js

```diff
@@ -268,13 +286,26 @@ class App extends Component {
     this.setState({ isLoading: option });
   };
 
+  siteLookup(site) {
+    const { sites } = this.state;
+    if (sites) {
+      for (let i = 0; i < sites.length; i++) {
```

Comment:
I believe a for loop is more performant and you can continue using this with my blessing, if my blessing is something you seek. But did you know there's many ES6 alternate methods for iterables called "filter" that does the same thing? There's also the for..of loop

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

Reply:
This might fail, but something along the lines of 

```js
siteLookup({key: siteId = false}) {
  return siteId && this.state.sites && this.state.sites.filter(site => site.id === siteId)[0]
}
```
Similarly, there is the for..of loop:

```js
const { sites } = this.state
if (!sites) return
for (let iteratedSite of sites) {
 if (site.key === iteratedSite.id) return iteratedSite;
}
return false

```